grails-app/services/au/org/ala/ecodata/SpatialService.groovy

```diff
@@ -76,12 +77,26 @@ class SpatialService {
         }
 
         start = end
-        filterOutObjectsInBoundary(result, geo)
-        end = System.currentTimeMillis()
-        log.info("Time taken to filter out objects in boundary: ${end-start}ms")
 
-        start = end
-        Map geographicFacets = convertResponsesToGeographicFacets(result)
+        Map geographicFacets
+        if (geo.geometryType == 'GeometryCollection') {
+            geographicFacets = [:].withDefault{[]}
+            GeometryCollection geometryCollection = (GeometryCollection)geo
+            for (int i=0; i<geometryCollection.numGeometries; i++) {
+
+                Map filtered = filterOutObjectsInBoundary(result, geometryCollection.getGeometryN(i))
+                start = end
+                Map geographicFacetsForGeometry = convertResponsesToGeographicFacets(filtered)
+                geographicFacetsForGeometry.each { k, v ->
+                    geographicFacets[k] += v
+                    geographicFacets[k] = geographicFacets[k].unique()
+                }
+            }
+        }
+        else {
+            Map filtered = filterOutObjectsInBoundary(result, geometryCollection.getGeometryN(i))
```

Comment:
Looks like `geometryCollection`  and `i` are not defined here.

Reply:
Thanks @temi I've fixed this.